README.md

```diff
@@ -1,11 +1,314 @@
 
-## Running Tests
+This contract is intended for Decentraland's LAND and Estates but can be used with any ERC721 that has a `setUpdateOperator` function to give operator permissions to an address. For both of these assets, the update operator is the one that can deploy scenes on them.
+
+There are various ways to initialize a rent. What all of them have in common is that a user has to send a transaction to initialize the rent using a signature created off-chain by another user as verification of the rental conditions.
+
+## Index
+
+- [Listing](#listing)
+- [Accepting a Listing](#accepting-a-listing)
+- [Offers](#offers)
+- [Accepting an Offer](#accepting-an-offer)
+- [Claiming back the Asset](#claiming-back-the-asset)
+- [Changing the Update Operator of the Asset](#changing-the-update-operator-of-the-asset)
+- [What happens when the rental ends?](#what-happens-when-the-rental-ends)
+- [Extending a rental](#extending-a-rental)
+- [Nonces](#nonces)
+- [Invalidating Signatures](#invalidating-signatures)
+- [Development](#development)
+
+## Listing
+
+If I wanted to rent my LAND in order to accrue an extra income from it, I could do so by:
+
+<a id="authorize-rentals-contract"></a>
+1 - Authorizing the Rentals contract to operate LAND
+
+You can do so by calling `approve` on the LAND contract with the address of the Rentals contract and the token id of the LAND you want to rent.
+
+You can also do it with `setApprovalForAll` on the LAND contract with the address of the Rentals contract.
+
+The difference between one and the other is that you will need to call `approve` for each individual LAND you want to rent, while with `setApprovalForAll` you only have to do it once for all your assets in the contract.
+
+2 - Sign the rental conditions
+
+There are various conditions that you can set for a rental to occur. You can have a glimpse of what these conditions are by looking at the Listing struct.
+
+<a id="listing-struct"></a>
+```
+struct Listing {
+    address signer;
+    address contractAddress;
+    uint256 tokenId;
+    uint256 expiration;
+    uint256[3] nonces;
+    uint256[] pricePerDay;
+    uint256[] maxDays;
+    uint256[] minDays;
+    address target;
+    bytes signature;
+}
+```
+
+- signer: The address of the signer, in this case, my address.
+- contractAddress: The address of the LAND contract.
+- tokenId: The token id of the LAND I want to rent.
+- expiration: The timestamp from which the signature will not be valid anymore.
+- nonces: The three of them have to match the current nonces in the Rentals contract to be valid. Find more about it in the [Nonces](#nonces) section.
+- pricePerDay: Defines the price per day for the min and max days range in the same index.
+- maxDays: max amount of days the LAND can be rented for a given index.
+- minDays: min amount of days the LAND can be rented for a given index.
```

Comment:
this is a has?
```suggestion
- minDays: min amount of days the LAND has to be be rented for a given index.
```

meaning, can it be rented for less than `minDays`?

Reply:
No it can't